api.planx.uk/modules/flows/publish/service/applicationTypes.ts

```diff
@@ -0,0 +1,51 @@
+import {
```

Comment:
A couple quick reactions here: 
- I think this function could live within `flows/publish/helpers` - I think the `service/` subdirectory in addition to the existing `service.ts` file is quite confusing organisationally
- I'd like to see simple unit tests for this helper function _in addition to_ the full route-level `supertest` tests you've already written (eg `flows/publish/helpers.test.ts`). That would help us more easily/clearly test some of those cases you mention above like checking _other_ component types
- Can we chat tomorrow about how we might optimise this function and testing it?